src/controllers/llmo/llmo.js

```diff
@@ -387,6 +387,69 @@ function LlmoController(ctx) {
     }
   };
 
+  /**
+   * Checks if config changes are only AI-origin categorization updates.
+   * Returns true if all new/modified categories and topics contain only AI-origin prompts.
+   */
+  const areChangesAICategorizationOnly = (oldConfig, newConfig) => {
+    if (!oldConfig) return false;
+
+    const oldCategories = oldConfig?.categories || {};
+    const newCategories = newConfig?.categories || {};
+    const oldTopics = oldConfig?.topics || {};
+    const newTopics = newConfig?.topics || {};
+
+    // Get new category IDs
+    const newCategoryIds = Object.keys(newCategories).filter((id) => !oldCategories[id]);
+
+    // Get new or modified topic IDs
+    const changedTopicIds = Object.keys(newTopics).filter((id) => {
+      if (!oldTopics[id]) return true; // New topic
```

Comment:
how are these IDs computed? are they just based on the topic value? 

asking because i wonder if we can get false negatives if an old topic is removed and a new topic is assigned the same ID